### DISCUSSION TITLE
Support Figma's localVariables

### DISCUSSION THREAD

**Discussion by hayawata3626** [mikaelvesavuori/figmagic#196] [Q&A] [answered]
This is wonderful tool! 
Does this plugin support Figma's [localVariables](https://www.figma.com/plugin-docs/working-with-variables/)? If it doesn't, are there any plans to support localVariables in the future?

**Comment by mikaelvesavuori** [comment, marked as answer]
Hey @hayawata3626!

I am happy that you like Figmagic.

The new Local Variables concept in Figma is really exciting and it fills one of the problems Figma has had since day 1: The issue of design tokens not being supported _internally_ within Figma. This is also part of why I've written in the README, articles, and courses on how Figmagic and Figma are somewhat different when it comes to design tokens. To use Figmagic, you had (and still have) to do a bit of extra work to comply with the specific pages that Figmagic expects if you also have lots of Figma styles. There's a fair bit of double work and wrangling to keep them updated if you do both actual design and tokens.

Previously, in Figma, you had to resort to only using styles as there were no other systematic ways to decompose design. Styles would however not easily translate across more complex use cases. With Local Variables, at least internally in Figma, you can work with _de facto_ design tokens - very cool! The feature looks great.

However, the feature is still in beta and the [only docs I can find that refer to an API](https://www.figma.com/plugin-docs/working-with-variables/) mention it being available through the Plugin API, which is a completely different API than the one used by Figmagic (the "REST" API). I have also confirmed that the local variables do not transport across the REST API.

Even if there was support from the Figma side, it may be hard (or easy) to support depending entirely on their data model. The requirement would be to have a concise, authoritative set of these tokens.

While Figmagic has been growing slowly over the years, I have yet to see Figma itself go deep in developer-facing tooling - even almost 6 years after opening up their API we are only recently seeing some movement in this space. That's why Figmagic keeps being relevant to developers and designers, but it always assumes Figma makes new features available through some channel, such as their APIs. I also don't see Figmagic becoming a Plugin because that would negate the automation capabilities that are so central to Figmagic.

I don't have very high expectations that we'll get a clear answer to the above questions in the near term. 

Hope that sets a bit of context for why the answer is "no".
